Client/mods/deathmatch/logic/lua/CLuaFunctionDefs.World.cpp

```diff
@@ -1920,3 +1922,147 @@ int CLuaFunctionDefs::FetchRemote(lua_State* luaVM)
     lua_pushboolean(luaVM, false);
     return 1;
 }
+
+// table getRemoteRequests([resource theResource = nil])
+int CLuaFunctionDefs::GetRemoteRequests(lua_State* luaVM)
+{
+    CScriptArgReader argStream(luaVM);
+    CResource*       pResource = nullptr;
+    CLuaMain*        pLuaMain = nullptr;
+    int              iIndex = 0;
+
+    argStream.ReadUserData(pResource, NULL);
+
+    // Grab virtual machine
+    if (pResource)
+        pLuaMain = pResource->GetVM();
+
+    lua_newtable(luaVM);
+    for (const auto& request : g_pClientGame->GetRemoteCalls()->GetCalls())
+    {
+        if(!pLuaMain || request->GetVM() == pLuaMain)
+        {
+            lua_pushnumber(luaVM, ++iIndex);
+            lua_pushuserdata(luaVM, request);
+            lua_settable(luaVM, -3);
+        }
+    }
+
+    return 1;
+}
+
+// table getRemoteRequestInfo(element theRequest[, number postDataLength = 0])
+int CLuaFunctionDefs::GetRemoteRequestInfo(lua_State* luaVM)
+{
+    CScriptArgReader argStream(luaVM);
+    CLuaArguments    info, requestedHeaders;
+    CRemoteCall*     pRemoteCall = nullptr;
+    CResource*       pThisResource = g_pClientGame->GetResourceManager()->GetResourceFromLuaState(luaVM);
+    int              iPostDataLength = 0;
+
+    argStream.ReadUserData(pRemoteCall);
+    argStream.ReadNumber(iPostDataLength, 0);
+
+    if (!argStream.HasErrors())
+    {
+        CResource* pResource = pRemoteCall->GetVM()->GetResource();
+        bool bExtendedInfo = (pResource == pThisResource);
+        
+        info.PushString("type");
+        info.PushString((pRemoteCall->IsFetch() ? "fetch" : "call"));
+
+        // remove query_string from url when bExtendedInfo isn't set
+        SString sURL = pRemoteCall->GetURL();
+
+        if (!bExtendedInfo)
+            sURL = sURL.ReplaceI("%3F", "?").Replace("#", "?").SplitLeft("?");
+
+        info.PushString("url");
+        info.PushString(sURL);
+
+        info.PushString("queue");
+        info.PushString(pRemoteCall->GetQueueName().c_str());
+        
+        info.PushString("resource");
+        
+        if (pResource)
+            info.PushResource(pResource);
+        else
+            info.PushBoolean(false);
+
+        info.PushString("start");
+        info.PushNumber(pRemoteCall->GetStartTime());
+
+        if (bExtendedInfo)
+        {
+            SString sPostData = pRemoteCall->GetOptions().strPostData.c_str();
```

Comment:
I think this may cause issues with binary post data. Is there a specific reason why you use `c_str()`  here?

Reply:
Must've gotten in there when i tested something. Gonna clean that up in the next few hours.

Reply:
done @sbx320. Felt kinda useless to have an options table. If we ever decide on expanding options we might switch to options tables.